src/components/hooks/useMembers.js

```diff
@@ -34,8 +34,8 @@ export default function useMemberships(destinationID, destinationType) {
       // eslint-disable-next-line no-console
       console.error(error.message);
     };
-    const onMembers = (data) => {
-      setMembers([...data.members]);
+    const onMembers = (membersList) => {
```

Comment:
just `members` is fine

Reply:
@cipak, I would like to keep the `onMember` function as this is the only way the component can guarantee the state stays immutable. The component cannot know if the adapter implementation has mutated the original member array or created a new one, so we just make sure it's always a new one before updating the state. We ran into this issue in the past and it was tricky to find. Although we fixed in in the SDK adapter, we can prevent others extending the adapter from shooting themselves in the foot.

Reply:
@lalli-flores I see. Yes, immutability helps with such issues. I see that all hooks use this approach, and provide copies instead of the original data received from the sdk. However, these copies are shallow, so someone could still break something if they modify one of the elements in this array. I guess it's a trade-off, deep copies would be too expensive, right?

Reply:
@cipak, correct, deep copies are too expensive. Shallow copies are ok in this case, because we only want to make sure that React understand this is a new state by giving it a new array object (memory address), even if the contents of the array are still pointing to the previous objects.